.github/workflows/js_tests.yml

```diff
@@ -0,0 +1,272 @@
+name: JS tests master
+
+on:
+  push:
+    branches:
+      - master
+      - 3.x-maintenance
+    paths:
+      - '.github/scripts/*.sh'
+      - '.github/workflows/*.yml'
+      - 'gui/package.json'
+      - 'requirements/*.txt'
+      - 'Makefile'
+      - 'MANIFEST.in'
+      - '**.bat'
+      - '**.jsx?'
+      - '**.sh'
+
+  pull_request:
+    branches:
+      - master
+      - 3.x-maintenance
+    paths:
+      - '.github/scripts/*.sh'
+      - '.github/workflows/*.yml'
+      - 'gui/package.json'
+      - 'requirements/*.txt'
+      - 'MANIFEST.in'
+      - '**.bat'
+      - '**.jsx?'
+      - '**.sh'
+
+jobs:
+  ############################################################################
+  # Close previous builds on the same branch
+  ############################################################################
+  cleanup-previous-runs:
+    name: Cleanup previous builds
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo $GITHUB_EVENT_NAME
+      - uses: rokroskar/workflow-run-cleanup-action@v0.2.2
+        if: github.event_name == 'pull_request'
+        env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+
+  ############################################################################
+  # Lint jobs
+  ############################################################################
+  lint-js:
+    name: Lint JS
+    needs: cleanup-previous-runs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Will this be set when it is not a PR? Will it now checkout the correct commit without this ref spec?

Reply:
The ref in a Pr points to the merge commit this just makes it point to the actual last commit. Nothing will not eork because of this

Reply:
I could add a conditional if you prefer to be sure 🤷

Reply:
I just want to know it works :laughing:. If you say it does, sure let's keep it.

Reply:
Nope actually it does not work, will make the push for the fix later today.